selfdrive/controls/lib/longitudinal_mpc_lib/long_mpc.py

```diff
@@ -347,15 +344,13 @@ def update_with_xva(self, x, v, a):
     for i in range(N):
       self.solver.cost_set(i, "yref", self.yref[i])
     self.solver.cost_set(N, "yref", self.yref[N][:COST_E_DIM])
-    self.accel_limit_arr[:,0] = -10.
-    self.accel_limit_arr[:,1] = 10.
+    accel_limit_arr = np.full((N+1, 2), (-10., 10.))
```

Comment:
Let's only build it once since it's static.

Reply:
Since we only change the prev_a live parameter I just set all the other parameters once in the `set_weights` function